app/models/user.rb

```diff
@@ -26,4 +29,8 @@ def name
   def send_devise_notification(notification, *args)
     devise_mailer.send(notification, self, *args).deliver_later
   end
+
+  def generate_uuid
+    self.unique_user_reference = SecureRandom.hex(5) if unique_user_reference.blank?
```

Comment:
You'd need to add some logic here for checking if the reference has already been taken.

E.g. 

```ruby
  return if unique_user_reference.present?
  while unique_user_reference.blank?
    ref = SecureRandom.hex(5) 
    self.unique_user_reference = ref if User.where(unique_user_reference: ref).blank?
  end 
```

Reply:
this is what I was after, thank you!